serialization/base.py

```diff
@@ -0,0 +1,469 @@
+    ):
+        """
+        :param code_object_path: e.g.`i6_experiments.users.username.some_experiment.pytorch_networks.SomeNiceASRModel`.
+            This can be the object itself, e.g. a function or a class. Then it will use __qualname__ and __module__.
+        :param unhashed_package_root: The root path to a package, from where relatives paths will be hashed.
+            Recommended is to use the root folder of an experiment module. E.g.:
+            `i6_experiments.users.username.some_experiment`
+            which could be retrieved via `__package__` from a module in the root of the `some_experiment` folder.
+            In case one wants to avoid hash conflicts this might cause, passing an `ExplicitHash` object to the
+            same collection as the import is possible.
+        :param import_as: if given, the code object will be imported as this name
+        :param use_for_hash: if False, this import is not hashed when passed to a Collection/Serializer
+        :param ignore_import_as_for_hash: do not hash `import_as` if set
+        """
+        super().__init__()
+        if not isinstance(code_object_path, str):
+            assert getattr(code_object_path, "__qualname__", None) and getattr(code_object_path, "__module__", None)
+            mod_name = code_object_path.__module__
+            qual_name = code_object_path.__qualname__
+            assert "." not in qual_name
+            assert getattr(sys.modules[mod_name], qual_name) is code_object_path
+            code_object_path = f"{mod_name}.{qual_name}"
+        self.code_object = code_object_path
+
+        self.object_name = self.code_object.split(".")[-1]
+        self.module = ".".join(self.code_object.split(".")[:-1])
+        self.package = ".".join(self.code_object.split(".")[:-2])
+
+        if unhashed_package_root:
+            if not self.code_object.startswith(unhashed_package_root):
+                raise ValueError(
+                    f"unhashed_package_root: {unhashed_package_root} is not a prefix of {self.code_object}"
+                )
+            self.code_object = self.code_object[len(unhashed_package_root) :]
+
+        self.import_as = import_as
+        self.use_for_hash = use_for_hash
+        self.ignore_import_as_for_hash = ignore_import_as_for_hash
+
+    def get(self) -> str:
+        """get. this code is run in the task"""
+        if self.import_as:
+            return f"from {self.module} import {self.object_name} as {self.import_as}\n"
+        return f"from {self.module} import {self.object_name}\n"
+
+    def _sis_hash(self):
+        if self.import_as and not self.ignore_import_as_for_hash:
+            return sis_hash_helper({"code_object": self.code_object, "import_as": self.import_as})
+        return sis_hash_helper(self.code_object)
+
+    def __hash__(self):
+        if self.import_as and not self.ignore_import_as_for_hash:
+            return hash({"code_object": self.code_object, "import_as": self.import_as})
```

Comment:
maybe I am understanding something wrong.. but this would be the builtin hash func, yes? AFAIK that does not take dicts as an arg.. so is this `short_hash` or am I missing something?

Reply:
```suggestion
            return short_hash({"code_object": self.code_object, "import_as": self.import_as})
```